### PR TITLE
[FEATURE] changes for multidomainsetup: TCA, Flexforms, pageTS, documentation

### LINKED ISSUE #122: Selectable categories with multiple domains
Hello,

I am using the seminars extension for multiple websites in one TYPO3 installation. After creating a seminar on the first website, all categories of the project are shown - also the ones from other websites, which is not what I want.
It would be nicer, if there was one folder, that contained the categories, organizers/hosts and locations and could further be browsed in recursively.
Or is there already a solution for this out there, that I am not seeing?

Thanks a lot in advance!
Cheers,

### PR DESCRIPTION
**Pull request by leofuehringer**
I rebased it and edited the DefaultControllerTest.php
Sorry for the confusion with github. I'm learning it ;-)
This PR fixes #122

### CODE REVIEW

**Comment by leofuehringer** [comment]
wohooo... it looks good.
I hope that it was ok, that I deleted the lines in the DefaultControllerTest.php...

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 ## x.y.z
 
 ### Added
+- changes for multidomainsetup: TCA, Flexforms, pageTS, documentation
```
Comment (oliverklee) [inline CHANGELOG.md]: Please add a link to the PR ID.

Also, I suggest a wording change. I propose: `Multi-domain setup`

File: Configuration/TCA/tx_seminars_seminars.php [hunk 2]
```diff
@@ -116,7 +116,7 @@
                 'internal_type' => 'db',
                 'allowed' => 'tx_seminars_seminars',
                 'foreign_table' => 'tx_seminars_seminars',
-                'foreign_table_where' => 'AND tx_seminars_seminars.uid <> ###THIS_UID### AND object_type = 1 ORDER BY title',
+                'foreign_table_where' => 'AND tx_seminars_seminars.pid <> ###THIS_UID### AND object_type = 1 ORDER BY title',
```
Comment (oliverklee) [inline Configuration/TCA/tx_seminars_seminars.php]: Why the change `uid => pid`? This doesn't look right to me.

File: Tests/Unit/FrontEnd/DefaultControllerTest.php [hunk 3]
```diff
@@ -1210,10 +1210,6 @@ public function testSingleViewForSeminarWithoutImageDoesNotDisplayImage()
         );
         $this->fixture->piVars['showUid'] = $this->seminarUid;
 
-        self::assertNotContains(
-            'style="background-image:',
-            $this->fixture->main('', [])
-        );
```
Comment (oliverklee) [inline Tests/Unit/FrontEnd/DefaultControllerTest.php]: This change does not belong in this PR.

File: Configuration/TSconfig/ContentElementWizard.txt [hunk 4]
```diff
@@ -11,3 +11,22 @@ mod.wizards.newContentElement.wizardItems.plugins {
         }
     }
 }
+
+#Marker for the Configuration/TCA/tx_seminars_seminars.php
```
Comment (oliverklee) [inline Configuration/TSconfig/ContentElementWizard.txt]: After the `#` sign, please add a space (also for the other comment).

Comment (oliverklee) [inline Configuration/TSconfig/ContentElementWizard.txt]: Please let's describe this more in terms of what these lines do, not to which files they relate.

File: Configuration/TSconfig/ContentElementWizard.txt [hunk 5]
```diff
@@ -11,3 +11,22 @@ mod.wizards.newContentElement.wizardItems.plugins {
...
+TCEFORM.tx_seminars_seminars.categories.PAGE_TSCONFIG_STR = %
```
Comment (oliverklee) [inline Configuration/TSconfig/ContentElementWizard.txt]: Please let's find a name for that variable that describes more what it does. `PAGE_TSCONFIG_STR` only describes where it is and its type (and also, it's an abbreviation).

File: Configuration/TSconfig/ContentElementWizard.txt [hunk 6]
```diff
@@ -11,3 +11,22 @@ mod.wizards.newContentElement.wizardItems.plugins {
...
+TCEFORM.tx_seminars_seminars.event_type.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.place.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.lodgings.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.foods.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.speakers.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.partners.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.tutors.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.leaders.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.organizers.PAGE_TSCONFIG_STR = %
+TCEFORM.tx_seminars_seminars.organizing_partners.PAGE_TSCONFIG_STR = %
```
Comment (oliverklee) [inline Configuration/TSconfig/ContentElementWizard.txt]: Please let's remove the redundancies by using a structure like this:
```
TCEFORM {
  tx_seminars_seminars {
    leaders.PAGE…
    organizers.PAGE…
  }
}
```

File: Documentation/DE/Konfiguration/MultidomainEinrichten/Index.rst [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+.. role::   typoscript(code)
+.. role::   ts(typoscript)
+   :class:  typoscript
+.. role::   php(code)
+
+
+Multidomain einrichten
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Bei der Verwendung von Seminars für mehrere Domains in einem TYPO3 Projekt erscheinen standardmäßig alle Kategorien, Zahlungsarten, Orte, Veranstalter, etc., die irgendwo im Projekt zu finden sind. Werden nun mehrere Homepages in einem Projekt betreut, können sowohl die anzuzeigenden Einträge, sowohl in der Bearbeitung eines Seminars, als auch in der Flexform beim Anlegen eines Seiteninhaltselements eingeschränkt werden.
```
Comment (oliverklee) [inline Documentation/DE/Konfiguration/MultidomainEinrichten/Index.rst]: TYPO3 Projekt -> TYPO3-Projekt

File: Documentation/DE/Konfiguration/MultidomainEinrichten/Index.rst [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+   TCEFORM.tx_seminars_seminars.organizing_partners.PAGE_TSCONFIG_STR =384
+   TCEFORM.tx_seminars_seminars.target_groups.PAGE_TSCONFIG_STR =384
+   TCEFORM.tx_seminars_seminars.checkboxes.PAGE_TSCONFIG_STR =384
+   TCEFORM.tx_seminars_seminars.payment_methods.PAGE_TSCONFIG_STR =384
+   
+   TCEFORM.tt_content.pi_flexform.PAGE_TSCONFIG_STR=384
+
+
+
+
```
Comment (oliverklee) [inline Documentation/DE/Konfiguration/MultidomainEinrichten/Index.rst]: Please drop the surplus blank lines.

File: Documentation/EN/Configuration/SetupMultidomain/Index.rst [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+.. role::   underline
+.. role::   typoscript(code)
+.. role::   ts(typoscript)
+   :class:  typoscript
+.. role::   php(code)
+
+
+Setup Multidomain
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you use seminars for multiple domains, all categories, payment methods, places, etc. existing in the typo3 project are shown by default. With the following configurations you can restrict the displayed entries in editing a seminar as well in using the flexforms.
```
Comment (oliverklee) [inline Documentation/EN/Configuration/SetupMultidomain/Index.rst]: typo3 -> TYPO3

Comment (oliverklee) [reply Documentation/EN/Configuration/SetupMultidomain/Index.rst]: configurations -> configuration

File: Documentation/EN/Configuration/SetupMultidomain/Index.rst [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+   TCEFORM.tx_seminars_seminars.organizing_partners.PAGE_TSCONFIG_STR =384
+   TCEFORM.tx_seminars_seminars.target_groups.PAGE_TSCONFIG_STR =384
+   TCEFORM.tx_seminars_seminars.checkboxes.PAGE_TSCONFIG_STR =384
+   TCEFORM.tx_seminars_seminars.payment_methods.PAGE_TSCONFIG_STR =384
+   
+   TCEFORM.tt_content.pi_flexform.PAGE_TSCONFIG_STR=384
+
+
+
+
```
Comment (oliverklee) [inline Documentation/EN/Configuration/SetupMultidomain/Index.rst]: Please drop the surplus blank lines.

**Comment by oliverklee** [comment]
Closing, as there has been no feedback for quite some time. @leofuehringer, please feel free to reopen this PR if you are going to work on it again. Thanks for contributing!
